src/polkadot/PolkadotEngine.ts

```diff
@@ -210,6 +243,116 @@ export class PolkadotEngine extends CurrencyEngine<
     this.addTransaction(this.currencyInfo.currencyCode, edgeTransaction)
   }
 
+  async queryLiberlandTransactions(): Promise<void> {
+    return await queryTxMutex(
+      async () => await this.queryLiberlandTransactionsInner()
+    )
+  }
+
+  async queryLiberlandTransactionsInner(): Promise<void> {
+    let hasNextPage = true
+    let endCursor = null
+    let processedCount = 0
+
+    const endpoint = this.networkInfo.liberlandScanUrl
+    if (endpoint == null) {
+      this.warn('`liberlandBaseUrl` not defined')
+      return
+    }
+
+    const userId = this.walletInfo.keys.publicKey
+
+    while (hasNextPage) {
+      const operationName = 'Transfers'
+      const query = `query Transfers($userId: String, $cursor: Cursor) {
+        transfers(
+          filter: {
+            or: [
+              { fromId: { equalTo: $userId } },
+              { toId: { equalTo: $userId } }
+            ]
+          },
+          after: $cursor,
+          first: 50,
+          orderBy: BLOCK_NUMBER_DESC
+        ) {
+          nodes {
+            id
+            fromId
+            toId
+            value
+            eventIndex
+            block {
+              number
+              timestamp
+            }
+          }
+          pageInfo {
+            hasNextPage
+            endCursor
+          }
+          totalCount
+        }
+      }`
+
+      const variables = {
+        userId,
+        cursor: endCursor
+      }
+
+      try {
+        const response = await this.fetchLiberlandScan(
+          endpoint,
+          operationName,
+          query,
+          variables
+        )
+        const cleanResponse = asLiberlandTransfersResponse(response)
+
+        const transfers = cleanResponse.data.transfers.nodes
+
+        transfers.forEach(tx => {
+          const edgeTransaction = processLiberlandTransaction(
+            {
+              walletId: this.walletId,
+              walletInfo: this.walletInfo,
+              currencyInfo: this.currencyInfo,
+              allTokensMap: this.allTokensMap
+            },
+            tx
+          )
+          if (edgeTransaction != null)
+            this.addTransaction(this.currencyInfo.currencyCode, edgeTransaction)
+        })
+
+        hasNextPage = cleanResponse.data.transfers.pageInfo.hasNextPage
+        endCursor = cleanResponse.data.transfers.pageInfo.endCursor
+
+        // Update progress
+        const totalCount = cleanResponse.data.transfers.totalCount
+        processedCount += transfers.length
+
+        this.tokenCheckTransactionsStatus[this.currencyInfo.currencyCode] =
+          totalCount === 0 ? 1 : processedCount / totalCount
+        this.updateOnAddressesChecked()
+      } catch (e: any) {
+        this.warn(`Error fetching Liberland transactions: ${e.message}`)
+        break
```

Comment:
Breaking here would exit the while loop and set tokenCheckTransactionsStatus to 1 which wouldn't be accurate.

Reply:
Issue still exists. throw the errors instead of breaking so you don't update the sync status

Reply:
Somehow didn't push my actual changes, sorry about that